manubot/cite/isbn.py

```diff
@@ -11,32 +11,41 @@ def get_isbn_citeproc(isbn):
     Currently, uses Citoid to retrieve metadata, although which resource this
     function delegates to may change in the future.
     """
-    return get_isbn_citeproc_citoid(isbn)
+    import isbnlib
+    isbn = isbnlib.to_isbn13(isbn)
+    for retriever in isbn_retrievers:
+        try:
+            return retriever(isbn)
+        except Exception as error:
+            logging.warning(
+                f'Error in {retriever.__name__} for {isbn} '
+                f'due to a {error.__class__.__name__}:\n{error}'
+            )
+            logging.info(error, exc_info=True)
+    raise Exception(f'all get_isbn_citeproc methods failed for {isbn}')
 
 
-def get_isbn_citeproc_isbnlib(isbn):
+def get_isbn_citeproc_zotero(isbn):
     """
-    Generate CSL JSON Data for an ISBN using isbnlib.
+    Generate CSL JSON Data for an ISBN using Zotero's translation-server.
     """
-    import isbnlib
-    metadata = isbnlib.meta(isbn, cache=None)
-    csl_json = isbnlib.registry.bibformatters['csl'](metadata)
-    csl_data = json.loads(csl_json)
-    return csl_data
+    from manubot.cite.zotero import export_as_csl, search_query
+    zotero_data = search_query(f'isbn:{isbn}')
+    csl_data = export_as_csl(zotero_data)
+    csl_item, = csl_data
+    return csl_item
 
 
 def get_isbn_citeproc_citoid(isbn):
     """
     Return CSL JSON Data for an ISBN using the Wikipedia Citoid API.
     https://en.wikipedia.org/api/rest_v1/#!/Citation/getCitation
     """
-    import isbnlib
     import requests
     from manubot.util import get_manubot_user_agent
     headers = {
         'User-Agent': get_manubot_user_agent(),
     }
-    isbn = isbnlib.to_isbn13(isbn)
```

Comment:
There are only two types of ISBNs: ISBN10 and ISBN13. I think all the methods we have work with both types, but we probably want to use ISBN13, since it is the newer identifier system meant to support a larger number of books. I moved this passthrough to the get_isbn_citeproc rather than having it in every function.